commands/cdns.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2018 The Doctl Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"fmt"
+
+	"github.com/digitalocean/doctl"
+	"github.com/digitalocean/doctl/do"
+	"github.com/digitalocean/godo"
+	"github.com/spf13/cobra"
+)
+
+// CDN creates the CDN command
+func CDN() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:   "cdn",
+			Short: "cdn commands",
+			Long:  "cdn is used to access cdn commands",
+		},
+	}
+
+	CmdBuilder(cmd, RunCDNList, "list", "list cdn", Writer,
+		aliasOpt("ls"), displayerType(&cdn{}))
+
+	cmdCDNCreate := CmdBuilder(cmd, RunCDNCreate, "create <cdn-origin>", "create a cdn", Writer,
+		aliasOpt("c"), displayerType(&cdn{}))
+	AddIntFlag(cmdCDNCreate, doctl.ArgCDNTTL, "", 3600, "CDN ttl")
+
+	cmdRunCDNDelete := CmdBuilder(cmd, RunCDNDelete, "delete <cdn-id>", "delete a cdn", Writer,
+		aliasOpt("rm"))
+	AddBoolFlag(cmdRunCDNDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Force cdn delete")
+
+	CmdBuilder(cmd, RunCDNGet, "get <cdn-id>", "get a cdn", Writer, aliasOpt("g"),
+		displayerType(&cdn{}))
+
+	cmdCDNUpdate := CmdBuilder(cmd, RunCDNUpdateTTL, "update <cdn-id>", "update a cdn", Writer,
+		aliasOpt("u"), displayerType(&cdn{}))
+	AddIntFlag(cmdCDNUpdate, doctl.ArgCDNTTL, "", 0, "cdn ttl",
+		requiredOpt())
+
+	cmdCDNFlushCache := CmdBuilder(cmd, RunCDNFlushCache, "flush <cdn-id>", "flush cdn cache", Writer,
+		aliasOpt("fc"))
+	AddStringSliceFlag(cmdCDNFlushCache, doctl.ArgCDNFiles, "", []string{}, "cdn files",
```

Comment:
Is the empty slice a valid default here? The `godo` docs indicate that the argument should be non-empty.

Reply:
You're correct, it's not a valid choice. The endpoint will return an error. But the `--files` flag is a required flag, so the default should not be called since a `doctl` error will be raised if the user doesn't include the `--files` flag.

Reply:
Ah, right, I missed that it's required. Given that wildcards are allowed `[]string{"*"}` might be a reasonable default, but keeping it required makes sense too.

Reply:
Yeah I thought about that and had it as the wildcard at first, I was just wary of assuming the User's intention, though flushing the cache isn't the end of the world. I agree that `*` would make a good default. I'll update that.